### DISCUSSION TITLE
selecting metrics from "stdmetrics"

### DISCUSSION THREAD

**Discussion by taeyoonlee87** [r-lidar/lidR#674] [Q&A] [answered]
Hi, 

I am trying to create the equation using selected "stdmetrics" variables.  

As far as I known, it is not possible to select certain stdmetrics variables (such as zq5, pzabove2, zpcum6,,,,).

Could you please guide me on how I can define each variable manually?

I feel like I can define these variables like 

zmean = mean(z),
zsd = sd(z),
zq5 = quantil(z,c(0.05), and so on. 

Even I checked the description of each variable (from here: https://github.com/r-lidar/lidR/wiki/stdmetrics), I have no idea how I can express 'zpcum', 'p1th', 'pabove2'. in R code.

Thanks in advance,

**Comment by Jean-Romain** [comment, marked as answer]
You can copy the source code.

https://github.com/r-lidar/lidR/blob/master/R/metrics_stdmetrics.R#L123

Or ou can simply compute all metrics and select the ones of interest.

```r
myMetrics = function(Z)
{
  ans = stdmetrics(Z)
  ans = ans[c("zmean", "zsd", "...")]
  return(ans)
}
```

**Reply by taeyoonlee87** [reply]
I always appreciate your responses!
